transform/mattermost-analytics/models/marts/product/nps/fct_nps_feedback.sql

```diff
@@ -0,0 +1,16 @@
+select nf.server_id as server_id
+       , nf.user_id as user_id
+       , nf.license_id as license_id
```

Comment:
It's tricky to retain license ID here. Let's take as an example the case for adding license information to the current feedback. The following sub-cases exist:
- License ID is null. This means that a join is required between fact and dim table to get the last available license information `<=` the feedback's date.
- License ID is not null and a single row for that license id exists in the dimension. A join on server id and license id is required.
- License ID is not null and multiple rows for that license id exists in the dimension. A join on server id and license id is required, but it will require deduplication (perhaps using feedback date?).
- License ID is not null but no row exists in license id.

Tackling the above cases will require a rather challenging query. The user of the dimensional model will also need to know all these details. Otherwise it's tempting to just join on columns with the same name, resulting in potential bugs due to all the cases described above.

IMHO the dimensional model needs to handle all these cases and offer a simplified model for the developer.